### DISCUSSION TITLE
How to feed VAD detection results to diarization pipeline

### DISCUSSION THREAD

**Discussion by kafan1986** [pyannote/pyannote-audio#844] [Q&A]
I have already performed VAD step using a different model. If required, I can create a Pynote.Audio.Annotation object from my VAD data. I believe this data more or less represent the segmentation stage output. Is my understanding correct? 

Now question is how can I feed this object to embedding and clustering step for performing diarization? I am using the develop branch currently.

**Comment by hbredin** [comment]
There is no easy way to feed your VAD into the existing diarization pipeline as the latter does not explicitely rely on a VAD step. 

You'd have to design your own speaker diarization pipeline. You can use [PretrainedSpeakerEmbedding](https://github.com/pyannote/pyannote-audio/tree/develop#speaker-verification) to extract embeddings and then [any clustering algorithm from scikit-learn](https://scikit-learn.org/stable/modules/clustering.html) for instance.
